rosbag2_cpp/src/rosbag2_cpp/writers/sequential_writer.cpp

```diff
@@ -97,6 +97,16 @@ void SequentialWriter::open(
 
   const auto storage_uri = format_storage_uri(base_folder_, 0);
 
+  rcpputils::fs::path db_path(storage_uri);
```

Comment:
why not using `base_folder_`? That should avoid the need to verify the `parent_path` of it.